python/lsst/ts/dsm/dsm_csc.py

```diff
@@ -214,28 +188,36 @@ def process_yaml_file(self, ifile):
           The filename to read and process.
         """
         self.log.info(f"Process {ifile} file.")
-        with open(os.path.join(self.telemetry_directory, ifile), 'r') as infile:
+        with open(os.path.join(self.telemetry_directory, ifile), "r") as infile:
             content = yaml.safe_load(infile)
-            ui_config_file = pathlib.PosixPath(content['ui_versions']['config_file']).as_uri()
-            self.tel_configuration.set_put(dsmIndex=self.salinfo.index,
-                                           timestampConfigStart=utils.convert_time(content['timestamp']),
-                                           uiVersionCode=content['ui_versions']['code'],
-                                           uiVersionConfig=content['ui_versions']['config'],
-                                           uiConfigFile=ui_config_file,
-                                           cameraName=content['camera']['name'],
-                                           cameraFps=content['camera']['fps'],
-                                           dataBufferSize=content['data']['buffer_size'],
-                                           dataAcquisitionTime=content['data']['acquisition_time'])
+            ui_config_file = pathlib.PosixPath(
+                content["ui_versions"]["config_file"]
+            ).as_uri()
+            self.tel_configuration.set_put(
```

Comment:
Out of scope, but please consider (eventually) making this an event instead of a telemetry topic. Surely it is does not need to be output at frequent intervals? If it can change over time, but only sporadically, then an event is still a good choice. You can poll it and set the values as often as you like, and the event will only be output if any field changes.

Reply:
OK. I'll put it on the list to do on the next iteration since the next XML release is due out soon.